### ISSUE TITLE
Bug: `LambdaTimeout` error on insight page

### ISSUE THREAD

**Issue by jpmcb** [open-sauced/app#2420]
### Describe the bug

I can't always get this to replicate, but I hit this timeout semi-frequently on an insight page

![Screenshot 2024-01-04 at 2 04 00 PM](https://github.com/open-sauced/app/assets/23109390/7669b559-e281-4367-aefa-e67d90b565d1)

It appears to be a Netlify edge function timing out. Not sure what could be causing it.

Cc @brandonroberts also saw this happening

### Steps to reproduce

1. Go to an insight page with many members and repos in it

(again, difficult to replicate)

### Browsers

Firefox

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Contributing Docs

- [X] I agree to follow this project's Contribution Docs

**Comment by nickytonline** [comment]
It's a Netlify Function (not Netlify Edge Function) that is timing out. Netlify Functions have a 10 second execution limit.

I'm on mobile, but if we're retrieving data server-side that isn't basic Insight data like the name, we should offload that data retrieval to the client-side.

You can look in the function logs for the given ID to get more information about what timed out.

**Comment by jpmcb** [comment]
Yeah, I think it's because we're getting _alot_ of data for insight pages (especially with multiple members in a page). The fix is probably 2 fold:

- Optimize the query to not left join and select all rows for all members of an insight page
- Offload the data loading like you said

**Comment by nickytonline** [comment]
I can move the data fetching to the client when I'm back on Monday unless someone else gets to it first. It shouldn't be a big lift.
